lib/factor/factorQuadratic.js

```diff
@@ -222,44 +224,55 @@ function factorSumProductRule(node, symbol, aValue, bValue, cValue, negate) {
     const factorPairs = ConstantFactors.getFactorPairs(product, true);
     for (const pair of factorPairs) {
       if (pair[0] + pair[1] === bValue) {
-        // bValue is split into pair[0] + pair[1]
-        // group the terms
-        const firstGroup = [aValue, pair[0]];
-        const secondGroup = [cValue, pair[1]];
-        const firstGroupGcd = math.gcd(...firstGroup);
-        const secondGroupGcd = math.gcd(...secondGroup);
-
-        // factor the first group: e.g. 2x^2 + 4x -> 2x * (x + 2)
-        const firstCoeffValue = Node.Creator.constant(firstGroupGcd);
-        const firstCoeff = Node.Creator.polynomialTerm(
-          symbol, null, firstCoeffValue);
-
-        const firstTermInGroupCoeff = Node.Creator.constant(aValue/firstGroupGcd);
-        const firstTermInGroup = Node.Creator.polynomialTerm(
-          symbol, null, firstTermInGroupCoeff);
-        const secondTermInGroupValue = pair[0]/firstGroupGcd;
-        const secondTermInGroup = Node.Creator.constant(secondTermInGroupValue);
-
-        // factor the second group: e.g. 3x + 3 -> 3(x + 1)
-        // Since (x + 1) is the same after factoring both groups, we only find it once
-        let secondCoeffValue = secondGroupGcd;
-        if (pair[1] < 0) {
-          secondCoeffValue = secondCoeffValue * -1;
+        // To factor, we go through some transformations (these should be actual
+        // substeps in the future)
+        // 1. Break apart the middle term into two terms using our factor pair (p and q):
+        //    e.g. ax^2 + bx + c -> ax^2 + px + qx + c
+        // 2. Consider the first two terms together and the second two terms
+        //    together (this doesn't require any actual change to the expression)
+        //    e.g. first group: [ax^2 + px] and second group: [qx + c]
+        // 3. Factor both groups separately
+        //    e.g first group: [ux(rx + s)] and second group [v(rx + x)]
+        // 4. Finish factoring by combining the factored terms through grouping:
+        //    e.g. (ux + v)(rx + s)
+        const p = pair[0];
```

Comment:
consider calling these pvalue and qValue just cause they're numbers and not nodes